src/core/simulation.jl

```diff
@@ -835,30 +835,26 @@ function update_parameter!(
     problem::OperationsProblem,
     sim::Simulation,
 ) where {T <: PSY.Component}
-    components = get_available_components(T, problem.sys)
-    initial_forecast_time = get_simulation_time(sim, get_simulation_number(problem))
-    horizon = length(model_time_steps(problem.internal.optimization_container))
-    for d in components
-        # RECORDER TODO: Parameter Update from forecast
-        # TODO: Improve file read performance
-        forecast = PSY.get_time_series(
-            PSY.Deterministic,
-            d,
-            get_data_label(param_reference);
-            start_time = initial_forecast_time,
-            count = 1,
-        )
-        ts_vector = IS.get_time_series_values(
-            d,
-            forecast,
-            initial_forecast_time;
-            len = horizon,
-            ignore_scaling_factors = true,
-        )
-        component_name = PSY.get_name(d)
-        for (ix, val) in enumerate(get_parameter_array(container)[component_name, :])
-            value = ts_vector[ix]
-            JuMP.set_value(val, value)
+    TimerOutputs.@timeit RUN_SIMULATION_TIMER "update_parameter!" begin
```

Comment:
Since this is a very specific update parameter from the forecast, as opposed to updating parameters from other results or cache. Can we make this more specific? or move it to the overall update parameters function?